File: src/main/java/com/example/examplemod/BlockExample.java
package com.example.examplemod;

        import net.minecraft.block.ITileEntityProvider;
        import net.minecraft.tileentity.TileEntity;

public class BlockExample extends Block implements ITileEntityProvider {

    public BlockGenerator(Builder builder) {
        super(builder);
    }

    @Override
    public boolean hasTileEntity() {
        return true;
    }

    @Override
    public TileEntity createNewTileEntity(IBlockReader world) {
        return new TileEntityExample();
    }

}

File: src/main/kotlin/com/example/examplemod/BlockExample.kt
package com.example.examplemod

import net.minecraft.block.ITileEntityProvider
import net.minecraft.tileentity.TileEntity

class BlockExample : Block(), ITileEntityProvider {

    constructor(builder: Builder) : super(builder) {}

    fun hasTileEntity(): Boolean {
        return true
    }

    fun createNewTileEntity(world: IBlockReader): TileEntity {
        return TileEntityExample()
    }

}
